File: app/src/main/java/mal/art/shopin/Activity/ProductsList.java
package mal.art.shopin.Activity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import java.util.ArrayList;
import java.util.List;
import mal.art.shopin.Adapter.ProductListAdapter;
import mal.art.shopin.Model.Product;
import mal.art.shopin.R;
import mal.art.shopin.ViewModel.ProductViewModel;

public class ProductsList extends AppCompatActivity implements ProductListAdapter.OnAddProductListener {

  public static final int REQUEST_CODE = 1;

  private RecyclerView recyclerView;

  private List<Product> productsList = new ArrayList<>();

  private ImageView opacityToBackgroundImage;

  private ProductViewModel productViewModel;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_products_list);
    recyclerView = findViewById(R.id.product_list_recycler_view);

    // should be done during graphic design
    opacityToBackgroundImage = findViewById(R.id.opacity_background_image);
    opacityToBackgroundImage.setImageAlpha(80);

    final ProductListAdapter adapter = new ProductListAdapter(this, this);
    recyclerView.setAdapter(adapter);
    recyclerView.setLayoutManager(new LinearLayoutManager(this));

    productViewModel = ViewModelProviders.of(this).get(ProductViewModel.class);
    productViewModel.getAllProducts().observe(this, products -> {
      adapter.setProducts(products);
    });
  }

  @Override
  protected void onStart() {
    super.onStart();
    View overlay = findViewById(R.id.product_list_layout);
    overlay.setSystemUiVisibility(View.SYSTEM_UI_FLAG_HIDE_NAVIGATION | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY | View.SYSTEM_UI_FLAG_FULLSCREEN);
  }

  public void goToProductsList(View view) {
    Intent productsListIntent = new Intent(view.getContext(), ProductsList.class);
    view.getContext().startActivity(productsListIntent);
  }

  @Override
  public void onProductClick(int position) {
    // Dokończ to wywoływanie okna do dodawania produktu do listy zakupów. REQUEST_CODE jest placeholderem
    // jak sprawdzić który produkt jest klikalny - czyli do czego robić odowłanie .get(position)
    //    productsList.get(position);
    //    Intent intent = new Intent(ProductsList.this, AddProductToShoppingList.class);
    //    startActivityForResult(intent, REQUEST_CODE);

    Toast.makeText(getApplicationContext(), "click", Toast.LENGTH_SHORT).show();
  }
}

File: app/src/main/java/mal/art/shopin/activity/ProductsList.kt
package mal.art.shopin.activity

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import mal.art.shopin.R
import mal.art.shopin.adapter.ProductListAdapter
import mal.art.shopin.adapter.ProductListAdapter.OnAddProductListener
import mal.art.shopin.model.Product
import mal.art.shopin.repository.ProductRepository
import mal.art.shopin.viewModel.ProductViewModel
import java.util.ArrayList

class ProductsList : AppCompatActivity(R.layout.activity_products_list), OnAddProductListener {

  private val productsList: MutableList<Product> = ArrayList()

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    val recyclerView = findViewById<RecyclerView>(R.id.product_list_recycler_view)

    val adapter = ProductListAdapter(this, this)

    recyclerView.adapter = adapter
    recyclerView.layoutManager = LinearLayoutManager(this)

    val productViewModel = ViewModelProviders.of(this).get(ProductViewModel::class.java)
    productViewModel.allProducts.observe(this, Observer { products: List<Product>? ->
      adapter.setProducts(products)
      productsList.addAll(products!!)
    })
  }

  fun goToProductsList(view: View) {
    val productsListIntent = Intent(view.context, ProductsList::class.java)
    view.context.startActivity(productsListIntent)
  }

  override fun onProductClick(position: Int) {
    val productName = productsList[position].productName

    val productCategory = productsList[position].productCategory
    val intent = Intent(this@ProductsList, AddProductToShoppingList::class.java)

    intent.putExtra("productNameToAdd", productName)
    intent.putExtra("productCategoryToAdd", productCategory)

    startActivityForResult(intent, PRODUCT_ADD_REQUEST_CODE)
  }

  override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
    super.onActivityResult(requestCode, resultCode, data)

    if (requestCode == PRODUCT_ADD_REQUEST_CODE && resultCode == Activity.RESULT_OK) {
      val product = Product(
        data!!.getStringExtra("productName"),
        data.getStringExtra("productCategory"),
        Integer.valueOf(data.getStringExtra("productQuantity")!!),
        data.getStringExtra("productUnit"),
        getString(R.string.default_shopping_status)
      )
      ProductRepository.insertToShoppingList(product)
      Log.d("PRODUKT ZAPISANY DO FB", "$product.productName $product.productQuantity $product.productUnit")
    }
  }

  companion object {
    const val PRODUCT_ADD_REQUEST_CODE = 1
  }
}